### PR TITLE
Editor: Improve cylinder gizmos

### LINKED ISSUE #10214: Allow dragging top/bottom of Cylinder Shape and CSG Cylinder individually
### Describe the project you are working on

A 3D game using blockouts with CSG.

### Describe the problem or limitation you are having in your project

Positioning and changing the height of CSG Cylinders is annoying. The same does apply to cylinder shapes.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I wish both the bottom and top had a handle, like CSG Box nodes! This would allow positioning a CSG Cylinder on a surface and then changing it's height without it also extending down. If you needed to move both sides at once equally, that should still be possible with ALT, like on control nodes and Box meshes /CSG's.

Related to https://github.com/godotengine/godot/pull/80278

This has been [suggested before](https://github.com/godotengine/godot-proposals/issues/9487#issuecomment-2046221367), but I couldn't find a dedicated proposal.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Like the Box3D helper handles I think.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around with a few more movements/resizes of the shape. But it is quite annoying.

### Is there a reason why this should be core and not an add-on in the asset library?

CSG Cylinder is a core node. It is not consistent with the CSG Box.

### PR DESCRIPTION
**Pull request by HolonProduction**
Closes godotengine/godot-proposals#10214

https://github.com/user-attachments/assets/26a8d065-68fb-4d15-a2aa-db6d069eebc2

### CODE REVIEW

**Review by SaracenOne — APPROVED** [review]
Tested and yes, I think this is a sensible improvement to the behaviour of the gizmo. It might be worth adding similar behaviour to the capsules for consistency in the future, but I think this is fine to merge. 👍

**Review by AThousandShips — APPROVED** [review]

File: editor/plugins/gizmos/gizmo_3d_helper.h [hunk 1]
```diff
@@ -50,6 +50,11 @@ class Gizmo3DHelper : public RefCounted {
 	String box_get_handle_name(int p_id) const;
 	void box_set_handle(const Vector3 p_segment[2], int p_id, Vector3 &r_box_size, Vector3 &r_box_position);
 	void box_commit_handle(const String &p_action_name, bool p_cancel, Object *p_position_object, Object *p_size_object = nullptr, const StringName &p_position_property = "global_position", const StringName &p_size_property = "size");
+
+	Vector<Vector3> cylinder_get_handles(const real_t &p_height, const real_t &p_radius);
```
[the remark this answers is not in the document]
Comment (HolonProduction) [reply editor/plugins/gizmos/gizmo_3d_helper.h:54]: Is this a style thing or does it have negative implications?

[the remark this answers is not in the document]
Comment (AThousandShips) [reply editor/plugins/gizmos/gizmo_3d_helper.h:54]: It won't have any negative effects, there's no reason to pass these types by reference unless you bind to them elsewhere, as there's no data that will be updated, and there's no benefit in performance

**Review by KoBeWi — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**